libraries/SITL/SIM_Vicon.cpp

```diff
@@ -204,6 +205,49 @@ void Vicon::update_vicon_position_estimate(const Location &loc,
             NULL, 0);
         msg_buf[msg_buf_index].time_send_us = time_send_us;
     }
+
+    // determine time, position, and angular deltas
+    uint64_t time_delta = now_us + time_offset_us - _prev_time_us;
+
+    Quaternion q_curr;                   // Rotation to current MAV_FRAME_BODY_FRD from MAV_FRAME_LOCAL_NED
+    q_curr.from_euler(roll, pitch, yaw); // Rotation to MAV_FRAME_LOCAL_NED from current MAV_FRAME_BODY_FRD
```

Comment:
sorry I didn't catch this earlier but I think we can directly use SITL's "quaternion" variable to get the current attitude as a quaternion.

Reply:
I could use sitl's attitude quaternion here, but then I have to rotate it by the camera's yaw angle that was already done earlier in the code on line 142.

I renamed the variables hopefully more to your liking?